client/extensions/woocommerce/state/sites/coupons/handlers.js

```diff
@@ -48,10 +60,54 @@ export function requestCouponsSuccess( { dispatch }, action, { data } ) {
 	dispatch( couponsUpdated( siteId, params, body, totalPages, totalCoupons ) );
 }
 
+export function couponCreate( { dispatch }, action ) {
+	const { siteId, coupon } = action;
+	const path = 'coupons';
+
+	dispatch( request( siteId, action ).post( path, coupon ) );
+}
+
+export function couponCreateSuccess( { dispatch }, action ) {
+	// TODO: Update local state for this coupon.
```

Comment:
I don't see this handled in any open PRs at the moment and it's not logged in the Isotope thread either. Just flagging to make sure it is done.